packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1025,6 +1064,43 @@ describe("Eth module", function () {
           );
         });
 
+        it("Should return the deployed code in the context of a new block with 'pending' block tag param", async function () {
+          const snapshotId = await this.provider.send("evm_snapshot");
+          const contractAddressBefore = await deployContract(
+            this.provider,
+            `0x${EXAMPLE_CONTRACT.bytecode.object}`
+          );
+
+          assert.isNotNull(contractAddressBefore);
+
+          const contractCodeBefore = await this.provider.send("eth_getCode", [
+            contractAddressBefore,
+            "latest",
+          ]);
+
+          await this.provider.send("evm_revert", [snapshotId]);
+          await this.provider.send("evm_setAutomineEnabled", [false]);
+
+          const txHash = await this.provider.send("eth_sendTransaction", [
+            {
+              from: DEFAULT_ACCOUNTS_ADDRESSES[0],
+              data: `0x${EXAMPLE_CONTRACT.bytecode.object}`,
+              gas: numberToRpcQuantity(DEFAULT_BLOCK_GAS_LIMIT),
+            },
+          ]);
+          const contractAddressAfter = await this.provider.send(
```

Comment:
This variable would actualy hold a tx receipt object. `contractAddress` is a field of tx recept. Let's rename it to `deployTxReceipt` and `contractAddressBefore` -> `contractAddress`.